plugins/modules/source_control/gitlab/gitlab_project_variable.py

```diff
@@ -164,27 +166,34 @@ def list_all_project_variables(self):
             vars_page = self.project.variables.list(page=page_nb)
         return variables
 
-    def create_variable(self, key, value, masked, protected, variable_type):
+    def create_variable(self, key, value, masked, protected, variable_type, environment_scope):
         if self._module.check_mode:
             return
-        return self.project.variables.create({"key": key, "value": value,
-                                              "masked": masked, "protected": protected,
-                                              "variable_type": variable_type})
-
-    def update_variable(self, key, var, value, masked, protected, variable_type):
-        if var.value == value and var.protected == protected and var.masked == masked and var.variable_type == variable_type:
+        var = {
+            "key": key, "value": value,
+            "masked": masked, "protected": protected,
+            "variable_type": variable_type
+        }
+        if environment_scope is not None:
+            var["environment_scope"] = environment_scope
+        return self.project.variables.create(var)
+
+    def update_variable(self, key, var, value, masked, protected, variable_type, environment_scope):
+        if (var.value == value and var.protected == protected and var.masked == masked
+                and var.variable_type == variable_type and var.environment_scope == environment_scope):
```

Comment:
```suggestion
                and var.variable_type == variable_type and (var.environment_scope == environment_scope or environment_scope is None)):
```
This makes sure that if `environment_scope` is not specified in the module arguments, the module will not care what the current value for that field is.